### ISSUE TITLE
Example : A dismissed Slidable widget is still part of the tree.

### ISSUE THREAD

**Issue by giorgio79** [wwwdata/implicitly_animated_reorderable_list#23]
Hello,

Fixed the example so it works with the latest Flutter :)

I am noticing that the Slideable widget when dismissed emits a small error. Please see video. Any tips how to make this issue go away?  : 
[dismissed widget still part of the tree error.webm](https://github.com/wwwdata/implicitly_animated_reorderable_list/assets/283347/ee9b58ed-46bf-42a1-92cd-abfbcbb67af7)

The recommended solution by Stackoverflow is to use UniqueKey instead of ValueKey https://stackoverflow.com/questions/55792521/how-to-fix-a-dismissed-dismissible-widget-is-still-part-of-the-tree-error-in

When  switch to that, I am getting a small flicker in animation, as both Dismissible and AnimatedList is doing some animation.

[double animation.webm](https://github.com/wwwdata/implicitly_animated_reorderable_list/assets/283347/93688cef-4515-4269-a478-02c667b6af38)

```
======== Exception caught by widgets library =======================================================
The following assertion was thrown building SlidableDismissal(state: _SlidableDismissalState#c7a52(ticker inactive)):
A dismissed Slidable widget is still part of the tree.


Make sure to implement the onDismissed handle of the ActionPane and to immediately remove the Slidable widget from the application once that handler has fired.

```

**Comment by giorgio79** [comment]
Stackoverflow to the rescue!!!! 
https://stackoverflow.com/questions/45919197/flutter-dismissible-insists-list-item-must-be-removed-from-tree
need to use confirmDismiss instead of onDismiss...

Testing the proposal right now

**Comment by giorgio79** [comment]
Still not perfect, but this one hopefully https://github.com/flutter/flutter/issues/29844

**Comment by wwwdata** [comment]
So what about the suggestion to set `removeDuration` to zero and work with this `confirmDismiss` callback?

**Comment by giorgio79** [comment]
Yes, tried that. For some reason I still see the flicker in the Implicitly Animated widget. I tested going back to simple AnimatedList and no flicker there. Probably gonna go with AnimatedList for now.

**Comment by UngNhoThach** [comment]
tks bro so much

**Comment by makoConstruct** [comment]
It seems like this is partly due to an [async yield happening in the didUpdateWidget](https://github.com/wwwdata/implicitly_animated_reorderable_list/blame/99bf2830efa9049f166340417505f593228eaff4/lib/src/implicitly_animated_list_base.dart#L163), meaning the animated list doesn't actually update on the same frame as the dismissible is removed (happens on the frame after). Investigating.

**Comment by makoConstruct** [comment]
It seems like there's no way to block on a future in dart, so fixing this would require removing the MyersDiff call (MyersDiff doesn't allow users to run the diff algorithm synchronously, that method is private)

didUpdateWidget/its caller is synchronous, so if you await anything, the awaited code (myersdiff) runs _after_ the widget tree is built, so the dismissed item will never be removed in time. (_though it seems like it actually does sometimes, for reasons I can't explain and which definitely aren't reliable_)
